server/test/durablejobs/jobs/CopyFileKeyForMultipleFileUploadTest.java

```diff
@@ -0,0 +1,461 @@
+    assertOldAndNewDataEquals(latestProgramFileUpload, "applicantFile.jpg");
+  }
+
+  @Test
+  public void run_doesntModifyUnansweredQuestions() {
+    ProgramModel program =
+        ProgramBuilder.newActiveProgram("program")
+            .withBlock("block1")
+            .withOptionalQuestion(testQuestionBank.fileUploadApplicantFile())
+            .build();
+    ProgramQuestionDefinition programFileUpload =
+        ProgramQuestionDefinition.create(
+            testQuestionBank.fileUploadApplicantFile().getQuestionDefinition(),
+            Optional.of(program.id));
+    ApplicantModel applicant = resourceCreator.insertApplicantWithAccount();
+    ApplicantData applicantDataBeforeJob = applicant.getApplicantData();
+
+    runJob();
+
+    applicant.refresh();
+    ApplicantQuestion latestProgramFileUpload =
+        new ApplicantQuestion(programFileUpload, applicant.getApplicantData(), Optional.empty());
+    assertOldAndNewDataUnanswered(latestProgramFileUpload);
+
+    assertThat(applicant.getApplicantData().asJsonString())
+        .isEqualTo(applicantDataBeforeJob.asJsonString());
+  }
+
+  @Test
+  public void run_doesntOverwriteNewFormat() {
+    ProgramModel program =
+        ProgramBuilder.newActiveProgram("program")
+            .withBlock("block1")
+            .withOptionalQuestion(testQuestionBank.fileUploadApplicantFile())
+            .build();
+    ProgramQuestionDefinition programFileUpload =
+        ProgramQuestionDefinition.create(
+            testQuestionBank.fileUploadApplicantFile().getQuestionDefinition(),
+            Optional.of(program.id));
+    ApplicantModel applicant = resourceCreator.insertApplicantWithAccount();
+
+    // This shouldn't happen anywhere but a staging environment, but make sure that if we have
+    // existing data in the new format, we don't overwrite it.
+    ApplicantQuestion applicantFileUploadQuestion =
+        new ApplicantQuestion(programFileUpload, applicant.getApplicantData(), Optional.empty());
+    QuestionAnswerer.answerFileQuestion(
+        applicant.getApplicantData(),
+        applicantFileUploadQuestion.getContextualizedPath(),
+        "oldData.jpg");
+    QuestionAnswerer.answerFileQuestionWithMultipleUpload(
+        applicant.getApplicantData(),
+        applicantFileUploadQuestion.getContextualizedPath(),
+        ImmutableList.of("newData.jpg"));
+
+    applicant.save();
+
+    runJob();
+
+    applicant.refresh();
+    ApplicantQuestion latestProgramFileUpload =
+        new ApplicantQuestion(programFileUpload, applicant.getApplicantData(), Optional.empty());
+    assertThat(latestProgramFileUpload.createFileUploadQuestion().getFileKeyValue().get())
+        .isEqualTo("oldData.jpg");
+    assertThat(latestProgramFileUpload.createFileUploadQuestion().getFileKeyListValue().get())
+        .containsExactly("newData.jpg");
+  }
+
+  @Test
+  public void
+      run_copiesFileKeyForApplicantAndApplicationsWithEnumeratorAndNonEnumeratorQuestions() {
```

Comment:
**suggestion:** May be overkill, but could be worth testing with a nested enumerator that has file upload as well.

Reply:
I'm not sure what that is so yeah happy to test it (I'm sure this approach works even not knowing that that is given how we're parsing the JSON, but a test is the only way to actually be sure :)

Any examples for tests that create nested enumerators?

Reply:
It's basically one level deeper - so an enumerator within an enumerator. For example, maybe there will be a program that has an enumerator `List all members of your household` and then there would be an enumerator within it `List all jobs that $this has had in the past year` (with $this as the enumerated entity above) and then a file upload within that one that says `Upload docs for that person's job` or something like that.

https://github.com/civiform/civiform/blob/main/server/test/support/TestQuestionBank.java#L572 should be an example of a nested enumerator setup. It basically is another enumerator question definition that sets the enumerator ID of the parent enum

Reply:
Ah actually I think I got it looking at the code, just an enumerator in an enumerator. Working on it now. Probably is overkill but eh!